lib/gooddata/rest/rest_aggregator.rb

```diff
@@ -0,0 +1,56 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require 'pathname'
+
+base = Pathname(__FILE__).dirname.expand_path
+Dir.glob(base + '*.rb').each do |file|
+  require_relative file
+end
+
+module GoodData
+  module Rest
+    class Aggregator
+      attr_reader :store
+
+      def initialize
+        @store = {}
+      end
+
+      def clear_store
+        @store.clear
+      end
+
+      def update_store(stat)
+        domain = stat[:domain].to_sym
+        method = stat[:method].to_sym
+        duration = stat[:duration]
+        endpoint = stat[:endpoint].to_sym
+        @store[domain] = {} unless @store.key?(domain)
+        @store[domain][method] = {} unless @store[domain].key?(method)
+        # puts @store[domain][method].key?(endpoint)
+        if @store[domain][method].key?(endpoint)
+          record = @store[domain][method][endpoint]
+          record[:min] = [duration, record[:min]].min
```

Comment:
Refers to 'record' more than self (maybe move it to another class?) (FeatureEnvy)